packages/components/src/Truncate/Truncate.js

```diff
@@ -1,111 +1,91 @@
-    }
-
-    render() {
-
-        const truncateClasses = classNames(
-            'ins-c-truncate',
-            this.props.className,
-            { [`is-inline`]: this.props.inline },
-            { [`is-block`]: !this.props.inline }
-        );
-
-        const trimmedText = this.props.text.substring(0, this.props.length);
-
-        const textOverflow = this.props.text.length > this.props.length;
-
-        const { showText } = this.state;
-
-        const expandButton =
-            <Button
-                className='ins-c-expand-button'
-                variant='link'
-                onClick={ this.toggleText }>
-                { this.props.expandText }
-            </Button>;
-
-        const collapseButton =
-            <Button
-                className='ins-c-collapse-button'
-                variant='link'
-                onClick={ this.toggleText }>
-                { this.props.collapseText }
-            </Button>;
-
-        if (this.props.inline) {
-            return (
-                <React.Fragment>
-                    <span
-                        className={ truncateClasses }
-                        widget-type='InsightsTruncateInline'
-                        dangerouslySetInnerHTML={ this.dangerousHtml(showText === false ? `${trimmedText}${textOverflow ? '...' : '' }`
-                            : this.props.text) } />
-                    { textOverflow && (showText === false ? expandButton : collapseButton) }
-                </React.Fragment>
-            );
-        } else {
-            return (
-                <Stack className={ truncateClasses }>
-                    <StackItem>
-                        <span
-                            widget-type='InsightsTruncateBlock'
-                            dangerouslySetInnerHTML={ this.dangerousHtml(showText === false ? `${trimmedText}${textOverflow ? '...' : '' }`
-                                : this.props.text) } />
-                    </StackItem>
-                    { textOverflow && <StackItem className={this.props.spaceBetween && 'pf-u-mt-sm'}>
-                        { showText === false ? expandButton : collapseButton }
-                    </StackItem>
-                    }
-                </Stack>
-            );
+        setShowText(!showText);
+    };
+
+    const expandButton =
+        <Button
+            className='ins-c-expand-button'
+            variant='link'
+            onClick={ toggleText }>
+            { expandText }
+        </Button>;
+    const collapseButton =
+        <Button
+            className='ins-c-collapse-button'
+            variant='link'
+            onClick={ toggleText }>
+            { collapseText }
+        </Button>;
+    const textWithOverflow = showText === false ? `${trimmedText}${textOverflow ? '...' : '' }` : text;
+    const html = dangerousHtml(textWithOverflow);
+    const mouseOverHandler = expandOnMouseOver && {
+        onMouseEnter: () => setShowText(true),
```

Comment:
Using `toggleText` for these worked in the browser, but it did not yield proper test results. 🤷 not sure why.